Mage.Sets/src/mage/cards/f/FurnacePunisher.java

```diff
@@ -0,0 +1,91 @@
+package mage.cards.f;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfUpkeepTriggeredAbility;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.condition.InvertCondition;
+import mage.abilities.condition.common.PermanentsOnTheBattlefieldCondition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.keyword.MenaceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.common.FilterLandPermanent;
+import mage.game.Game;
+import mage.players.Player;
+
+import java.util.UUID;
+
+public class FurnacePunisher extends CardImpl {
+
+    private static final FilterLandPermanent filter = new FilterLandPermanent();
+
+    static {
+        filter.add(SuperType.BASIC.getPredicate());
```

Comment:
This filter is unnecessary as it is out of the scope of the relevant effect (which has its own identical copy of this filter) and is not used within this `FurnacePunisher` class

Reply:
done